DataBiosphere/leonardo
# [IA-4418] Allow creator of a runtime to view it, if they can view the workspace

__Jira ticket__: https://broadworkbench.atlassian.net/browse/IA-4415




## Summary

- Permit runtime creator to always perform reads on the runtime.
- Sam permission checks will fail if the Azure runtime errors during create. Terra-workspace-manager will clean up Sam's policy records following the Azure VM failing and deleting itself. A Leo record, thus orphaned, should still be readable by the creator, who may want to check its errors for an explanation.
- This fix allows users with Workspace Reader to access runtimes they created. This behavior is consistent with listRuntimesV2, which shows a user with Workspace Reader all runtimes in a workspace (creatorOnly off) or all runtimes they created in a workspace (creatorOnly on). The list endpoint does not authorize user access to any individual runtime via Sam, only the workspace, so it avoids the issue fixed here.

## Testing these changes

### What to test

- Create a runtime under conditions which will cause an error in Azure (you could create it in a billing project with quota limits or availability limits for your region, or you could manually interrupt the creation from the Azure Portal side so that terra-workspace-manager decides create has failed)
- Open the Cloud Environment modal for your failed runtime
- You should be able to view a meaningful error message whose ultimate source was Azure.



### Who tested and where

  - _NB: Rerun automation tests on this PR by commenting `jenkins retest` or `jenkins multi-test`._
